File: node/two.js
const math = require('mathjs');

const n = 1000;

const A = math.random([n, n]);

const start = Date.now();
const x = math.eigs(A).eigenvectors;
const end = Date.now();

console.log(((end - start) / 1000).toFixed(3));

File: cpp/two.cpp
#include <iostream>
#include <Eigen/Dense>
#include <chrono>

int main() {
	int n = 1000;

	Eigen::MatrixXd A = Eigen::MatrixXd::Random(n, n);

	auto start = std::chrono::steady_clock::now();
	Eigen::SelfAdjointEigenSolver<Eigen::MatrixXd> eigensolver(A);
	auto end = std::chrono::steady_clock::now();

	std::chrono::duration<double> time = end - start;
	std::cout << time.count();

	return 0;
}